lib/ruby_lsp/global_state.rb

```diff
@@ -104,8 +104,10 @@ def detect_test_library
       if direct_dependency?(/^rspec/)
         "rspec"
       # A Rails app may have a dependency on minitest, but we would instead want to use the Rails test runner provided
-      # by ruby-lsp-rails.
-      elsif direct_dependency?(/^rails$/)
+      # by ruby-lsp-rails. A Rails app doesn't need to depend on the rails gem itself, individual components like
+      # activestorage may be added to the gemfile so that other components aren't downloaded. Check for the presence
+      #  of bin/rails to support these cases.
+      elsif File.exist?(File.join(workspace_path, "bin/rails"))
```

Comment:
```suggestion
      elsif File.exist?(File.join(workspace_path, "bin", "rails"))
```

Reply:
Ah sorry I clicked merge at the same time 😅

Reply:
Let's apply this change. We can't use forward slashes for Windows.

Reply:
Hm, is that really the fix then? https://ruby-doc.org/3.3.0/File.html#method-c-join doesn't mention this being platform-aware and seems to always use `/` anyways.

Reply:
Then maybe we should be using `Pathname` instead and need to switch this to

```suggestion
      elsif Pathname.new(workspace_path).join("bin").join("rails").exist?
```